### PR TITLE
Integrating the first module and minor fixes

### PR DESCRIPTION
**Pull request by enricofoschi**

### CODE REVIEW

**Comment by enricofoschi** [comment]
I am aware that code coverage is super low initially. Tests will come in soon. Do not approve before it's 100%.

**Comment by enricofoschi** [comment]
wooohooo 100%!

**Comment by xunga** [comment]
Wooohoo! Can't wait to play with it!

File: src/workers/listener.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                        const commandRegExp = new RegExp(process.env.SLACK_BOT_NAME +
+                            ' ([:-a-z0-9]+)( [-a-z0-9]+)?( .*?)?$', 'i');
+                        const matches = text.match(commandRegExp).filter(match => match);
+
+                        // Loading Module
+                        const moduleName = matches[1];
+                        let moduleInstance = null;
+
+                        try {
+                            moduleInstance = require('../modules/' + moduleName + '.js').default;
```
Comment (adrianhunter) [inline src/workers/listener.js]: we don't need ``` + '.js' ```

File: test/mocks/mockery.js [hunk 2]
```diff
@@ -1,13 +1,26 @@
 import slackbots from './libraries/slackbots.js'
+import slackNode from './libraries/slack-node.js'
 import dotenv from 'dotenv';
+const parrot = require('./modules/parrot.js');
```
Comment (adrianhunter) [inline test/mocks/mockery.js]: let's stick to one approach here, and let's remove ".js" in all imports

File: test/workers/listener.js [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+        mockery.setup({
+            output: output,
+            input: input
+        });
+    });
+
+    it('Should initialize the listener', () => {
+        const myListener = listener({bot: slackBot()});
+        myListener.listen();
+        expect(myListener.listening).to.equal(true);
```
Comment (adrianhunter) [inline test/workers/listener.js]: this really feels like true === true, maybe improve the truthiness of the listener.listen function :-)

Comment (enricofoschi) [inline test/workers/listener.js]: Agree - quite useless.

**Pull request merged**